storm-client/src/jvm/org/apache/storm/cluster/StormClusterStateImpl.java

```diff
@@ -437,18 +436,14 @@ public void supervisorHeartbeat(String supervisorId, SupervisorInfo info) {
     public void workerBackpressure(String stormId, String node, Long port, long timestamp) {
         String path = ClusterUtils.backpressurePath(stormId, node, port);
         boolean existed = stateStorage.node_exists(path, false);
+        if (timestamp <= 0) {
```

Comment:
This check could be moved to the top of the function, if the timestamp is negative there's no reason to check if the znode exists.